src/PrestaShopBundle/Resources/config/routing/admin/improve/design/cms_pages.yml

```diff
@@ -1,78 +1,123 @@
-        redirectRoute: admin_cms_pages_index
-        redirectQueryParamsToKeep:
-          - 'id_cms_category'
+admin_cms_pages_search:
+  path: /search
+  methods: POST
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:search'
+    _legacy_controller: AdminCmsContent
 
-admin_cms_pages_create_cms_category:
-    path: /category/new
-    methods: [GET, POST]
-    defaults:
-        _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:createCmsCategory'
-        _legacy_controller: AdminCmsContent
+admin_cms_pages_category_create:
+  path: /category/new
+  methods: [GET, POST]
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:createCmsCategory'
+    _legacy_controller: AdminCmsContent
 
-admin_cms_pages_edit_cms_category:
-    path: /category/{cmsCategoryId}/edit
-    methods: [GET, POST]
-    defaults:
-        _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:editCmsCategory'
-        _legacy_controller: AdminCmsContent
-    requirements:
-        cmsCategoryId: \d+
+admin_cms_pages_category_edit:
+  path: /category/{cmsCategoryId}/edit
+  methods: [GET, POST]
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:editCmsCategory'
+    _legacy_controller: AdminCmsContent
+  requirements:
+    cmsCategoryId: \d+
 
-admin_cms_pages_delete_cms_category:
-    path: /category/{cmsCategoryId}/delete
-    methods: DELETE
-    defaults:
-        _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:deleteCmsCategory'
-        _legacy_controller: AdminCmsContent
-    requirements:
-        cmsCategoryId: \d+
+admin_cms_pages_category_delete:
+  path: /category/{cmsCategoryId}/delete
+  methods: DELETE
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:deleteCmsCategory'
+    _legacy_controller: AdminCmsContent
+  requirements:
+    cmsCategoryId: \d+
 
-admin_cms_pages_delete_bulk_cms_category:
-    path: /category/delete-bulk
-    methods: POST
-    defaults:
-        _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:deleteBulkCmsCategory'
-        _legacy_controller: AdminCmsContent
+admin_cms_pages_category_delete_bulk:
+  path: /category/delete-bulk
+  methods: POST
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:deleteBulkCmsCategory'
+    _legacy_controller: AdminCmsContent
 
-admin_cms_pages_toggle_cms_category:
-    path: /category/{cmsCategoryId}/toggle-status
-    methods: POST
-    defaults:
-        _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:toggleCmsCategory'
-        _legacy_controller: AdminCmsContent
-    requirements:
-        cmsCategoryId: \d+
+admin_cms_pages_category_toggle:
+  path: /category/{cmsCategoryId}/toggle-status
+  methods: POST
+  defaults:
+    _controller: 'PrestaShopBundle:Admin\Improve\Design\CmsPage:toggleCmsCategory'
+    _legacy_controller: AdminCmsContent
```

Comment:
Will you add the `_legacy_link` later ?

Reply:
yes got a todo added after whole page is migrated since another Pull request also has routes and I want to minimize rebasing as much as possible so I'll include _legacy_link to all routes with tests in freeze period I guess

Reply:
I get it, but without the `_legacy_link` the QA won't be able to test it from the menu right?
If you have the same routes in another PR is it possible that the last merged PR includes the `_legacy_link`, so that on freeze time the pages are fully accessible?

Reply:
We provide QA the URL and we create later a "Enable page X" PR which adds all legacy links, at this time QA checks them all 😉